migrate/indexerdb0to1.js

```diff
@@ -201,21 +219,14 @@ let count = 0;
 
   // Updating chainDB version to ensure
   // filter indexes are not duplicated.
-  const chainDB = bdb.create({
-    location: path.resolve(process.argv[2], '../chain'),
-    compression: true,
-    cacheSize: 32 << 20,
-    createIfMissing: false
-  });
+  const chain = path.resolve(process.argv[2], '../chain');
+  await updateChainDB(chain);
 
-  console.log('Opened %s.', path.resolve(process.argv[2], '../chain'));
-  await chainDB.open();
-
-  const chainVersion = await getVersion(chainDB);
-  if (chainVersion === 6) {
-    await updateChainVersion(chainDB, 7);
+  // update spvChainDB if it exists
+  const spvchain = path.resolve(process.argv[2], '../spvchain');
+  if (await fs.exists(spvchain)) {
```

Comment:
We may need to add the same check for full node `chain` directory as well. If you have only ever used bcoin in spv mode, there will not be a `~/.bcoin/chain` directory and the script will fail.

At the same time, the `migrate/latest` script also needs to be more clever, because if `index` doesn't exist we STILL need to run that script in order to bump the `chain` and/or `spvchain` databases.

Reply:
> At the same time, the `migrate/latest` script also needs to be more clever, because if `index` doesn't exist we STILL need to run that script in order to bump the `chain` and/or `spvchain` databases.

still needs this fix I think:

```
--> migrate/latest ~/.bcoin-spv-only
Opened /Users/matthewzipkin/.bcoin-spv-only/spvchain.
Already upgraded.
Migration complete.
Opened /Users/matthewzipkin/.bcoin-spv-only/wallet.
Checking version.
Updating wallets...
Updated 1 wallets.
Checking version.
Updating version to 8.
Migration complete.
--> bcoin --spv --prefix ~/.bcoin-spv-only
[info] (chain) Chain is loading.
[info] (chain) Checkpoints are enabled.
[info] (chaindb) Opening ChainDB...
Error: Database version mismatch for database: "chain". Please run a data migration before opening.
    at DB.verify (/Users/matthewzipkin/Desktop/work/bcoin/node_modules/bdb/lib/db.js:568:13)
    at async ChainDB.open (/Users/matthewzipkin/Desktop/work/bcoin/lib/blockchain/chaindb.js:65:5)
    at async Chain.open (/Users/matthewzipkin/Desktop/work/bcoin/lib/blockchain/chain.js:89:5)
    at async SPVNode.open (/Users/matthewzipkin/Desktop/work/bcoin/lib/node/spvnode.js:148:5)
    at async /Users/matthewzipkin/Desktop/work/bcoin/bin/spvnode:42:3
```

Reply:
Done!